inspirehep/ext/cache_manager/cache.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+import functools
+
+import logging
+
+from flask import current_app, request
+
+from flask_cache import Cache
+
+from invenio_base.globals import cfg
+
+logger = logging.getLogger(__name__)
+
+
+class InspireCache(Cache):
+    """Extends Cache from Flask Cache and adds mode-checking (debug or not) for memoize and cached methods.
+
+    The parameters in both inspire_memoize and inspire_cached are the same
+    as the parameters of the original methods in Flask Cache class
+
+    """
+
+    def inspire_memoize(self, timeout=None, make_name=None, unless=None):
+        def memoize(f):
+            @functools.wraps(f)
+            def decorated_function(*args, **kwargs):
+                if not cfg['DEBUG']:
```

Comment:
Here's an IPython session to prove this:
```python
In [1]: from invenio_base.globals import cfg

In [2]: from flask import current_app

In [3]: def from_cfg():
   ...:     return cfg['DEBUG']
   ...:

In [4]: def from_current_app():
   ...:     return current_app.config['DEBUG']
   ...:

In [5]: def from_flask():
   ...:     return current_app.debug
   ...:

In [6]: %timeit from_cfg()
The slowest run took 4.79 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 3: 4.58 µs per loop

In [7]: %timeit from_current_app()
The slowest run took 5.86 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 3: 3.25 µs per loop

In [8]: %timeit from_flask()
The slowest run took 5.13 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 3: 3.72 µs per loop
```
(yay microbenchmarks!)